mrp_production_request/models/stock_rule.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2018-19 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class StockOrder(models.Model):
+    _inherit = "stock.rule"
+
+    @api.multi
+    def _prepare_mrp_production_request(
+            self, product_id, product_qty, product_uom, location_id, name,
+            origin, values, bom):
+        self.ensure_one()
+        data = self._prepare_mo_vals(
+            product_id, product_qty, product_uom, location_id, name,
+            origin, values, bom)
+        data['state'] = 'to_approve'
+        orderpoint = values.get('orderpoint_id')
+        if orderpoint:
+            data['orderpoint_id'] = orderpoint.id
+        procurement_group = values.get('group_id')
+        if procurement_group:
+            data['procurement_group_id'] = procurement_group.id
+        return data
+
+    @api.multi
+    def _need_production_request(self, product_id):
+        return self.action == 'manufacture' \
+            and product_id.mrp_production_request
+
+    @api.multi
+    def _run_production_request(self, product_id, product_qty, product_uom,
+                                location_id, name, origin, values):
+        """Trying to handle this as much similar as possible to Odoo
+        production orders. See `_run_manufacture` in Odoo standard."""
+        request_obj = self.env['mrp.production.request']
+        request_obj_sudo = request_obj.sudo().with_context(
+            force_company=values['company_id'].id)
```

Comment:
Not blocking but probably we need to ensure than "company_id" key is in values ?

Reply:
This is mimicking https://github.com/odoo/odoo/blob/12.0/addons/mrp/models/stock_rule.py#L36
if it fails here, it would also fail for standard manufacturing orders and Odoo itself should have to fix it.

Reply:
OK, thanks